x-pack/packages/kbn-elastic-assistant/impl/knowledge_base/knowledge_base_settings_management/document_entry_editor.tsx

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiCheckbox,
+  EuiFieldText,
+  EuiForm,
+  EuiFormRow,
+  EuiMarkdownEditor,
+  EuiSuperSelect,
+  EuiIcon,
+  EuiText,
+} from '@elastic/eui';
+import React, { useCallback } from 'react';
+import { DocumentEntry } from '@kbn/elastic-assistant-common';
+import * as i18n from './translations';
+
+interface Props {
+  entry?: DocumentEntry;
+  setEntry: React.Dispatch<React.SetStateAction<Partial<DocumentEntry>>>;
+}
+
+export const DocumentEntryEditor: React.FC<Props> = React.memo(({ entry, setEntry }) => {
+  // Name
+  const setName = useCallback(
+    (e) => setEntry((prevEntry) => ({ ...prevEntry, name: e.target.value })),
+    [setEntry]
+  );
+
+  // Sharing
+  const setSharingOptions = useCallback(
+    (value) =>
+      setEntry((prevEntry) => ({
+        ...prevEntry,
+        users: value === i18n.SHARING_GLOBAL_OPTION_LABEL ? [] : undefined,
+      })),
+    [setEntry]
+  );
+  // TODO: KB-RBAC Disable global option if no RBAC
+  const sharingOptions = [
```

Comment:
nit: Consider memoizing this non-primitive value, because it's passed as a prop to a component below

Reply:
Will do! I will be extracting the whole sharing field into its own component in the next UI PR as it's used in both entry types.